src/main/groovy/org/shipkit/internal/notes/contributors/AllContributorsFetcher.java

```diff
@@ -40,16 +43,36 @@ ProjectContributorsSet fetchAllContributorsForProject(String repository, String
         return result;
     }
 
-    private Set<ProjectContributor> extractContributors(List<JsonObject> page, String readOnlyAuthToken) throws IOException, DeserializationException {
+    private Set<ProjectContributor> extractContributors(List<JsonObject> page, final String readOnlyAuthToken) throws IOException, DeserializationException {
         Set<ProjectContributor> result = new HashSet<ProjectContributor>();
-        for (JsonObject contributor : page) {
-            //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
-            //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
-            String url = (String) contributor.get("url");
-            GitHubObjectFetcher userFetcher = new GitHubObjectFetcher(url, readOnlyAuthToken);
-            JsonObject user = userFetcher.getPage();
-            result.add(GitHubAllContributorsJson.toContributor(contributor, user));
+        //Since returned contributor does not have 'name' element, we need to fetch the user data to get his name
+        //TODO add static caching of this. Names don't change that often, let's just cache this forever in build cache.
+        ExecutorService executor = Executors.newFixedThreadPool(4);
```

Comment:
It uses Java7 features. At the moment we still want to support Java6 for old projects that for various reasons cannot modernize Java version. How much complexity we need to add to support this solution if we remained Java6 compatible? We can revisit the Java6 compatibility requirement or decide to user Java7 as minimum required version to run Shipkit.